Ft google auth #187419170

MugemaneBertin2001 opened the pull request:
# what does this PR do?
- Help user to sign up and login with existing google account
# Description of Task to be completed:
- As a user I should be able to athenticate (register/login) via Google
# how to test it 
- go to ft-google-auth-#187419170 branch
- clone this repo 
- create and fill .env with local configuration
- run the app 
- go to browser : localhost:3000/login
- click on URL for login with google
- follow on screen instruction

IRUMVAEmmanuel1 committed:
ESLint Configuration PreCommit format and Test

MugemaneBertin2001 committed:
Feat(authentication): implementation of google authentication

- Add signin and sign up with google account
- return access token
- added test

delivers #187419170

teerenzo commented on .env.example:
```diff
@@ -1,4 +1,12 @@
+GOOGLE_CLIENT_ID="" use yours
+GOOGLE_CLIENT_SECRET="" use yours
+GOOGLE_CALLBACK_URL="" use yours
+
+CLOUDINARY_CLOUD_NAME= "" use yours
+CLOUDINARY_API_KEY= "" use yours
+CLOUDINARY_API_SECRET= "" use yours
```
we don't need this

MugemaneBertin2001 replied:
removed it

teerenzo commented on .vscode/settings.json:
```diff
@@ -0,0 +1,14 @@
+{
+    "sqltools.connections": [
+        {
+            "previewLimit": 50,
+            "server": "localhost",
+            "port": 5432,
+            "driver": "PostgreSQL",
+            "name": "new",
+            "group": "new",
+            "database": "ecommerce_db",
+            "username": "postgres"
+        }
+    ]
+}
```
please pay attention

MugemaneBertin2001 replied:
added .vscode in .gitignore and removed it

teerenzo commented on __test__/home.test.ts:
```diff
@@ -12,9 +12,15 @@ describe("Testing Home route", () => {
+  test("server should return status code of 200 --> given'/'", async () => {
     const response = await request(app).get("/");
 
+    expect(response.status).toBe(200);
+  }, 20000);
+  test("server should return status code of 200 and link to log in with google --> given'/login'", async () => {
+    const response = await request(app).get("/login");
+    expect(response.text).toBe("<a href='/auth/google'>click to here to Login</a>")
+
```
you should test what you have done not what you use to test it

MugemaneBertin2001 replied:
added tests for my endpoints

teerenzo commented on src/utils/server.ts:
```diff
@@ -3,16 +3,34 @@ import cors from "cors";
 app.use("/", homeRoute);
 app.use("/api/v1", appROutes);
 app.use("/docs", docRouter);
+app.use('/', googleRoutes)
```
remove this here,

MugemaneBertin2001 replied:
refined as requested

teerenzo commented:
PR name and description is wrong please check on naming convertion again

Heisjabo committed:
feat(edit-profile): Implement password update feature

   - Implement the feature to allow users update their password
   - Added validation check for old password, new password and confirm
     password
   - hash the new password before updating
   - documented the feature using swagger

[Delivers #187419174]

IRUMVAEmmanuel1 committed:
ESLint Configuration format

IRUMVAEmmanuel1 committed:
ESLint Configuration format

IRUMVAEmmanuel1 committed:
ESLint Configuration format

teerenzo commented:
you have many commits bertin squash them and are you sure you have changed 40 files
